respa_admin/views/resources.py

```diff
@@ -128,18 +138,18 @@ def post(self, request, *args, **kwargs):
     def forms_valid(self, form, unit_authorization_formset):
         self.object = form.save()
         unit_authorization_formset.instance = self.object
-        for formset in unit_authorization_formset.cleaned_data:
-            if 'subject' and 'level' in formset:
-                if formset['can_approve_reservation']:
-                    assign_perm('unit:can_approve_reservation', self.object, formset['subject'])
+        for form in unit_authorization_formset.cleaned_data:
+            if 'subject' and 'level' in form:
```

Comment:
I missed this earlier, but the if logic does not work as intended here.
Currently: `if ('subject') and ('level' in form):` `'subject'` is a string and always truthy.
Should be: `if 'subject' in form and 'level' in form:` properly checking both conditions.